xled/response.py

```diff
@@ -59,6 +59,9 @@ def data(self):
                     "No response to create application response data from"
                 )
 
+            if not self.response.ok:
+                raise ApplicationError(self.response.text)
```

Comment:
I'm not sure if raising ApplicationError() here is a good idea. If one would explicitly run `raise_for_status(propagate=True)` (see bellow) they would get `HTTPError()` exception. But here they would get ApplicationError for de facto HTTP error.

I guess better option would be to call here self.response.raise_for_status() and let underlying requests to raise `HTTPError()`.

What do you think?

Reply:
`raise_for_status` will raise if the response does not have `code` field in response JSON.
Unfortunately, the endpoint for querying configured timers does not include this field:
[1](https://xled-docs.readthedocs.io/en/latest/rest_api.html#get-timer), [2](https://github.com/timon/twinkly/blob/master/API.md#get-configured-timers)

HTTPError is a better choice, agreed.